sys/auto_init/saul/auto_init_adxl362.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2016 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization of ADXL362 driver
+ *
+ * @author      Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ *
+ * @}
+ */
+
+#ifdef MODULE_ADXL362
+
+#include "log.h"
+#include "saul_reg.h"
+
+#include "adxl362_params.h"
+#include "adxl362.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define ADXL362_NUMOF    (sizeof(adxl362_params) / sizeof(adxl362_params[0]))
+
+/**
+ * @brief   Allocation of memory for device descriptors
+ */
+static adxl362_t adxl362_devs[ADXL362_NUMOF];
+
+/**
+ * @brief   Memory for the SAUL registry temperature sensor entries
+ */
+static saul_reg_t saul_entries_temp[ADXL362_NUMOF];
+
+/**
+ * @brief   Memory for the SAUL registry acceleration sensor entries
+ */
+static saul_reg_t saul_entries_acc[ADXL362_NUMOF];
+
+/**
+ * @brief   Reference the driver structs.
+ * @{
+ */
+extern const saul_driver_t adxl362_temperature_saul_driver;
+extern const saul_driver_t adxl362_acceleration_saul_driver;
+/** @} */
+
+void auto_init_adxl362(void)
+{
+    for (unsigned int i = 0; i < ADXL362_NUMOF; i++) {
+        int res;
+        adxl362_t *dev = &adxl362_devs[i];
+        const adxl362_params_t *params = &adxl362_params[i];
+        res = spi_init_master(params->spi, SPI_CONF_FIRST_RISING, SPI_SPEED_1MHZ);
```

Comment:
This is an uglyness that will go away when #2528 is resolved (possibly by #4780)